### ISSUE TITLE
Error Incorrect timeZone Information provided

### ISSUE THREAD

**Issue by ray-holland-es** [facebook/hermes#846]
## Bug Description

![bug](https://user-images.githubusercontent.com/96197577/198671796-d62187e8-0001-473d-bb6b-c18d860a8caa.jpeg)

- [x] I have run `gradle clean` and confirmed this bug does not occur with JSC

A small percentage of our app users are getting app crashes because of "Error Incorrect timeZone Information provided". When we disable hermes we do not get this issue. It seems to be how timezone is returned.

We have been able to recreate the issue on iOS by manually setting the timezone to either "Oklahoma City" or "San Francisco"

Hermes version: whatever is built into RN 0.70.1
React Native version (if any): 0.70.1
OS version (if any): iOS 16.0 (and 15.x)
Platform (most likely one of arm64-v8a, armeabi-v7a, x86, x86_64):

## Steps To Reproduce

We have been able to recreate the issue on iOS by manually setting the timezone to "San Francisco" (probably others). Did this on a physical device by going to Settings -> General -> Date and Time. I'm not sure you can re-create on the iOS Simulator.

code example:

```
const timeZone = Intl.DateTimeFormat().resolvedOptions().timeZone; // For San Francisco this returns "US/Pacific"
const options = {
          day: '2-digit',
          month: '2-digit',
          year: '2-digit',
          timeZone: timeZone
        };
        
return new Intl.DateTimeFormat('en-US', options()).format(new Date(day)); // This crashes
```

## The Expected Behavior

App should not crash.

I'm not sure why it's returning "US/Pacific" - instead of "America/Los_Angeles".... also I would think that Intl.DateTimeFormat would take either timezone.

**Comment by fbmal7** [comment]
@ray-holland-es I tracked down the source of the error. On iOS, we are using [this API](https://developer.apple.com/documentation/foundation/nstimezone/1387223-knowntimezonenames) to get a list of valid timezones. Currently, US/Pacific is not considered 'up-to-date', and therefore it is not returned by Apple's API. The expected way to write that TZ is now "America/Los_Angeles". We are looking into possible solutions to the problem, but we have found the source of the problem at least; thanks for raising this issue.
